### PR TITLE
Cancel refactor

### PR DESCRIPTION
**Pull request by hcarneiro**
Using the new `.onCancelRequest()` and `.forwardCancelRequest()`

*Note:* the `for` loop is because each menu item creates a link provider.

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -178,6 +178,26 @@
     Fliplet.Studio.emit('reload-page-preview');
   });
 
+  Fliplet.Widget.onCancelRequest(function() {
+    console.log(menusPromises)
```
Comment (squallstar) [inline js/interface.js]: console.log

File: js/interface.js [hunk 2]
```diff
@@ -178,6 +178,26 @@
...
+    if (menusPromises) {
+      for (var key in menusPromises) {
```
Comment (squallstar) [inline js/interface.js]: is `for in` supported on all browsers?

File: js/interface.js [hunk 3]
```diff
@@ -178,6 +178,26 @@
...
+        // skip loop if the property is from prototype
+        if (!menusPromises.hasOwnProperty(key)) continue;
+
+        var obj = menusPromises[key];
+        for (var prop in obj) {
+          // skip loop if the property is from prototype
+          if(!obj.hasOwnProperty(prop)) continue;
+
+          obj[prop].forwardCancelRequest()
```
Comment (squallstar) [inline js/interface.js]: semicolon

File: js/interface.js [hunk 4]
```diff
@@ -178,6 +178,26 @@
...
+        }
+      }
+    }
+    if (!currentProvider) return;
+    currentProvider.forwardCancelRequest()
```
Comment (squallstar) [inline js/interface.js]: semicolon. also, previous row should use `{ return; }`. We don't use the inline block syntax.

File: js/interface.js (hunk 2 above)
Comment (hcarneiro) [reply js/interface.js]: @squallstar I suppose so

screen shot 2018-01-19 at 18 29 17

**Comment by hcarneiro** [comment]
@squallstar committed changes!

File: js/interface.js (hunk 2 above)
Comment (squallstar) [reply js/interface.js]: IE6, what a time to be alive!

**Pull request merged**